File: frontend/src/shared/types/stock.ts
export interface Stock {
  ticker: string;
  name: string;
  market: string;
}

export interface PricePoint {
  time: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
  change_percent: number;
  volume_change_percent: number;
}

export type PriceDirection = "up" | "down" | "flat";

export interface LivePrice {
  price: number;
  change: number;
  change_percent: number;
  direction: PriceDirection;
  open: number;
  high: number;
  low: number;
  volume: number;
}

export interface LivePriceResponse {
  available: boolean;
  as_of: string | null;
  live: LivePrice | null;
}

File: backend/app/schemas/stock.py
from typing import Literal

from pydantic import BaseModel


class Stock(BaseModel):
    ticker: str
    name: str
    market: str


class LivePrice(BaseModel):
    price: float
    change: float
    change_percent: float
    direction: Literal["up", "down", "flat"]
    open: float
    high: float
    low: float
    volume: int


class LivePriceResponse(BaseModel):
    available: bool
    as_of: str | None = None
    live: LivePrice | None = None


class PricePoint(BaseModel):
    time: str
    open: float
    high: float
    low: float
    close: float
    volume: int
    change_percent: float
    volume_change_percent: float


class StockPricesResponse(BaseModel):
    ticker: str
    prices: list[PricePoint]
